src/utilities/string-utils.ts

```diff
@@ -7,3 +7,8 @@ export const countWords = (text: string) => {
   const matches = text.match(/\b\w+\b/g); // transitions from non-word to word character.
   return matches ? matches.length : 0;
 };
+
+export const isImageUrl = (url: string): boolean => {
+  const imageUrlPattern = /^(https?:\/\/.*\.(gif|jpg|jpeg|png)|data:image\/(gif|jpg|jpeg|png);base64,)/i;
```

Comment:
I believe right now CLUE only supports PNG, but I put in the other file extensions in case that changes.